guardian/typerighter
# Prevent overlapping matches for the Regex matcher

## What does this change?

This PR removes overlapping matches from the Regex matcher results. Rules are registered in order. Matches produced by later rules supersede matches produced by earlier results.

It's likely we'll need to think carefully about how this might work across matchers, but for now this amends problematic behaviour for the current rule set.

Before:

<img width="895" alt="Screenshot 2020-08-12 at 07 41 17" src="https://user-images.githubusercontent.com/7767575/89983394-61e37a80-dc6f-11ea-89a9-cd5f474a5de2.png">

After:

<img width="895" alt="Screenshot 2020-08-12 at 07 40 19" src="https://user-images.githubusercontent.com/7767575/89983399-6445d480-dc6f-11ea-8747-b12fd978ee78.png">

## How to test

As an example, deploy to CODE, and run Typerighter in CODE Composer against the phrase, 'for many people today, especially for women, packaging and broadcasting your image is a readily monetizable skill'. 'Women' should contain a single match, not two matches, as was previously the case.

## How can we measure success?

We never see a regex rule overlapping another regex rule.
